lib/setup-terraform.js

```diff
@@ -119,11 +119,48 @@ credentials "${credentialsHostname}" {
   core.debug(`Adding credentials to ${credsFile}`);
   await fs.writeFile(credsFile, creds);
 }
+async function isVersionFileExist(versionFilePath) {
+  try {
+    await fs.access(versionFilePath);
+    return true;
+  } catch (err) {
+    return false;
+  }
+}
+async function resolveVersionInput() {
+  let version = core.getInput("terraform_version");
+  const versionFileInput = core.getInput("terraform_version_file");
+  if (version && versionFileInput) {
+    core.warning(
+      "Both terraform_version and terraform_version_file inputs are specified, only terraform_version will be used"
+    );
+  }
+  if (version) {
+    return version;
+  }
+  if (versionFileInput) {
+    const versionFilePath = path.join(
+      process.env.GITHUB_WORKSPACE,
+      versionFileInput
+    );
+    const fileExists = await isVersionFileExist(versionFilePath);
+    if (!fileExists) {
+      throw new Error(
+        `The specified terraform version file at: ${versionFilePath} does not exist`
+      );
+    }
+
+    version = await fs.readFile(versionFilePath, "utf-8");
```

Comment:
Note that this won't work if tfutils/tfenv#347 (allow whole line comments) is merged, and assuming of course that the user has the version file due to using `tfenv`.

I say this mainly because I think it's an important consideration for the (Hashicorp) team in making a product decision about it as mentioned above: `terraform` itself doesn't have a version file concept, so either:
1. `setup-terraform` would support a simple 'just the version' format, and Hashicorp owns that format;
1. it tries to support all formats seen in the wild, on the basis of user reports that certain things (like comments) break it or something (and, er, hopes they never conflict?);
1. it declares support for a specific format (e.g. `tfenv`'s) and aims to support that (and inherently then breaking changes in that format necessitate breaking changes in `setup-terraform`)